src/vec_int64_ppc.h

```diff
@@ -92,14 +100,752 @@
+  t2 = vec_muleuw ((vui32_t)vra, (vui32_t)vrb);
+#else
+  t2 = vec_mulouw ((vui32_t)vra, (vui32_t)vrb);
+#endif
+  t3 = vec_vmsumuwm ((vui32_t)vra, t1, z);
+  t4 = vec_vsld (t3, s32);
+  return (vui64_t)vec_vaddudm (t4, t2);
+}
+ * \endcode
+ * Which generates the following for POWER8:
+ * \code
+	addis   r9,r2,.rodata.cst16+0x60@ha
+	addi    r9,r9,.rodata.cst16+0x60@l
+	lxv     vs33,0,r9
+	vmulouw v13,v2,v3
+	vrld    v0,v3,v1
+	vmulouw v3,v2,v0
+	vmuleuw v2,v2,v0
+	vaddudm v2,v3,v2
+	vsld    v2,v2,v1
+	vaddudm v2,v13,v2
+ * \endcode
+ * \note The addition of <I>zeros</I> to the final sum of
+ * vec_vmsumuwm() (<I>vec_addudm (psum, vrc)</I>)has been optimized
+ * away by the compiler.
+ * This eliminates the xxspltib and one vaddudm instruction from the
+ * final code sequence.
+ *
+ * And we can assume that the constant load of <I>{ 32, 32 }</I>
+ * will be common-ed with other operations or hoisted out of
+ * loops. So the shift constant can be loaded early and vrld is not
+ * delayed.
+ * This keeps the POWER8 latency in the 19-28 cycle range.
+ *
+ * \section i64_endian_issues_0_0 Endian problems with doubleword operations
+ *
+ * From the examples above we see that the construction of higher
+ * precision multiplies requires significant massaging of input and
+ * output elements.
+ * Here merge even/odd, merge high/low, swap, and splat doubleword
+ * element operations are commonly used.
+ *
+ * PowerISA 2.06 VSX (POWER7) added the general purpose
+ * Vector Permute Doubleword Immediate (xxpermdi).
+ * The compiler generates some form of xxpermdi for the doubleword
+ * (double float, long int, bool long) merge/splat/swap operations.
+ * As xxpermdi's element selection is an immediate field, most
+ * operations require only a single instruction.
+ * All the merge/splat/swap doubleword variant are just a specific
+ * select mask value and the inputs to xxpermdi.
+ *
+ * Which is very useful indeed for assembling, disassembling, merging,
+ * splatting, swapping, and pasting doubleword elements.
+ *
+ * Of course it took several compiler releases to implement all the
+ * generic merge/splat/swap operations for the supported types.
+ * GCC 4.8 as the first to support vec_xxpermdi as a built-in.
+ * GCC 4.8 also supported the generic built-ins vec_mergeh, vec_mergel,
+ * and vec_splat for the vector signed/unsigned/bool long type.
+ * But endian sensitive vec_mergeh, vec_mergel, and vec_splat were not
+ * supported until GCC 7.
+ * And the generic vec_mergee, vec_mergeo built-ins where not supported
+ * until GCC 8.
+ *
+ * But as we have explained in \ref mainpage_endian_issues_1_1 and
+ * \ref i32_endian_issues_0_0 the little endian transforms applied by
+ * the compiler can cause problems for developers of multi-precision
+ * libraries. The doubleword forms of the generic merge/splat
+ * operations etc. are no exception. This especially annoying when the
+ * endian sensitive transformd are applied between releases of the
```

Comment:
s/transformd/transforms/

Reply:
yes the 's' and 'd' keys are next to each other.

Reply:
and fixed